packages/jsActions/mobile-resources-native/src/file-download/DownloadFile.ts

```diff
@@ -0,0 +1,85 @@
+// This file was generated by Mendix Studio Pro.
+//
+// WARNING: Only the following code will be retained when actions are regenerated:
+// - the import list
+// - the code between BEGIN USER CODE and END USER CODE
+// - the code between BEGIN EXTRA CODE and END EXTRA CODE
+// Other code you write will be lost the next time you deploy the project.
+import { Platform } from "react-native";
+import RNBlobUtil from "react-native-blob-util";
+import FileViewer from "react-native-file-viewer";
+import mimeTypes from "mime";
+
+// BEGIN EXTRA CODE
+function formatPath(...pathArgs: string[]): string {
+    return pathArgs.filter(arg => !!arg).join("/");
+}
+
+function sanitizeFileName(name: string) {
+    /* eslint-disable-next-line no-control-regex */
+    return name.replace(/[<>"?:|*/\\\u0000-\u001F\u007F]/g, "_");
+}
+
+async function getUniqueFilePath(path: string) {
+    if (!(await RNBlobUtil.fs.exists(path))) {
+        return path;
+    }
```

Comment:
This means that if the path already exists, we're going to access the file system for that path twice. I think it's okay to do the lightweight string manipulation first, even if it turns out to be unnecessary.

Reply:
✨